android/src/main/java/com/reactnativecommunity/netinfo/NetworkCallbackConnectivityReceiver.java

```diff
@@ -142,9 +148,13 @@ void updateAndSend() {
 
     private void asyncUpdateAndSend(int delay) {
         new Handler(Looper.getMainLooper()).postDelayed(() -> {
-            mCapabilities = getConnectivityManager().getNetworkCapabilities(mNetwork);
-            updateAndSend();
-
+            try {
+                mCapabilities = getConnectivityManager().getNetworkCapabilities(mNetwork);
+                updateAndSend();
+            } catch (SecurityException e) {
+                // Android 11 may throw a 'package does not belong' security exception here.
```

Comment:
```suggestion
                // Android 11 may throw a 'package does not belong' security exception here.
                // Google fixed Android 14, 13 and 12 with the issue where Chaland Jean patched those versions.
                // Android 11 is too old, so that's why we have to catch this exception here to be safe.
               //  https://android.googlesource.com/platform/frameworks/base/+/249be21013e389837f5b2beb7d36890b25ecfaaf%5E%21/
```